addons/project/models/project_task.py

```diff
@@ -1830,8 +1830,9 @@ def action_convert_to_subtask(self):
         }
 
     def action_archive(self):
-        self.filtered(lambda t: not t.display_in_project and t.parent_id).display_in_project = True
-        return super().action_archive()
+        child_tasks = self.child_ids.filtered(lambda child_task: not child_task.display_in_project)
+        (self.filtered(lambda t: not t.display_in_project and t.parent_id) + child_tasks).display_in_project = True
+        return super(Task, self + child_tasks).action_archive()
```

Comment:
We could recursively call `action_archive` for the subtasks because there is not only one level we could also have a subtask with another subtask as parent.
```suggestion
        child_tasks = self.child_ids.filtered(lambda child_task: not child_task.display_in_project)
        if child_tasks:
            child_tasks.action_archive()
        self.filtered(lambda t: not t.display_in_project and t.parent_id).display_in_project = True
        return super().action_archive()
```